test/regression.bzl

```diff
@@ -0,0 +1,46 @@
+"""Instantiate a regression test based on .py or .tcl
+files using resources in //test:regression_resources"""
+
+def _pop(kwargs, key, default):
+    """BUILD does not support kwargs, use None as a "kwargs at home" workaround"""
```

Comment:
bazel actually can go beyond kwargs. There are ways to declare what parameters are expected in a rule. I defer to @QuantamHD who knows all about this and can point you to the right documentation.

Reply:
This is a macro, not a rule, though.

Reply:
Yeah, I don't have any experience with whatever the stuff is called *.bzl files, I try to stay away from stuff that looks like Python :)

Reply:
There are two levels:

- macros(just a few def functions). This is pretty straightforward and what I do here.
- then there are rules(repository, normal rules) and module extensions. This second bit is powerful, fascinating and introduces many new concepts.

I don't think a rule would make the code more readable or improve the use case here. Maybe later if the use case becomes more sophisticated we could use something more than macros.

Reply:
yeah, sounds macro is the right approach.